lib/system/gc_common.nim

```diff
@@ -61,6 +61,39 @@ when defined(nimTypeNames):
       let (allocs, deallocs) = getMemCounters()
       c_fprintf(stdout, "[Heap] allocs/deallocs: %ld/%ld\n", allocs, deallocs)
 
+  type
+    HeapInstance* = object
+      name*: cstring
+      count*: int
+      sizes*: int
+    HeapDump* = object
+      instances*: seq[HeapInstance]
```

Comment:
It concerns me a bit that this causes further allocations, the very thing we seek to measure.

Reply:
Yes, thats why I tried to make the sequence reusable and only allocated a few times at most. I was using this procedure to repeatedly sample the heap using the same `var HeapDump` and at the end I only saw one instance.
 
```
        ...
        <type name="seq[HeapInstance]" count="1" sizes="1568"/>
</heap>
```